core/src/main/scala/cats/MonadDefer.scala

```diff
@@ -0,0 +1,21 @@
+package cats
+
+import simulacrum.typeclass
+import cats.Eval.always
+
+/**
+ * A [[Monad monad]] that allows for arbitrarily delaying the
+ * evaluation of an operation, triggering its execution on each run.
+ *
+ * @see [[ApplicativeEval]] for capturing effects in an `F[_]`
+ *     applicative context, but without the repeating side-effects
+ *     requirement.
+ */
+@typeclass trait MonadDefer[F[_]] extends Monad[F] with ApplicativeEval[F] {
```

Comment:
I agree on `Id` not having a `MonadDefer` instance. It wouldn't be really lazy and stackoverflow errors would happen (just got one while trying to use `Id` as my "simplest MonadDefer monad").